Js2

StianSto opened the pull request:
preliminary review: 
all the required features should have been added. some things like better UX and UI handling is needed. see projects for task yet to be completed before delivery.

StianSto commented:
Q: is the way i have added routing a good method, or is it unnecessary complicated?

Fermain commented on README.md:
```diff
@@ -1,2 +1,3 @@
 # SoMeApp_me-and-my-friends
-"Me &amp; My Friends" is a social media website where the user has more control over customization. set their own theme that appears on their banner when posting. This project is a project made das a school project.  
+
```
How do I run the project? How do I add to it? Which technologies does it use?

StianSto replied:
[RESOLVED] readme now contains detailed instructions on how to clone, install and run the app. it also includes other relevant info aobut the project.

Fermain commented on profile/edit/index.html:
```diff
@@ -0,0 +1,267 @@
+    <link rel="stylesheet" href="/dist/css/styles.css" />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <script src="/src/js/script.js" type="module"></script>
+    <script src="/node_modules/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
```
I think you should add `defer` to both of these.

StianSto replied:
you are right, ill add them promptly :)

Fermain commented on profile/edit/index.html:
```diff
@@ -0,0 +1,267 @@
+    <script src="/src/js/script.js" type="module"></script>
+    <script src="/node_modules/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
+  </head>
+  <body style="--sidebar-width-open: 40vw">
```
Why can't you do this in SCSS?

StianSto replied:
on my website i am using a side-bar for a friendlist. i am using the same sidebar but replacing the friendlist with an edit profile form on "edit profile". the width is different and unique for this page. this way seemed like the least amount of effort required to get the result i wanted. i could maybe change the variable in js?

Fermain commented on src/js/api/auth/loginAuth.mjs:
```diff
@@ -0,0 +1,45 @@
+
+/**
+ * user log in. needs valid email and password.
+ * @param {text} email valid email
```
```suggestion
 * @param {string} email valid email
```

Fermain commented on src/js/api/auth/loginAuth.mjs:
```diff
@@ -0,0 +1,45 @@
+/**
+ * user log in. needs valid email and password.
+ * @param {text} email valid email
+ * @param {text} password enter valid password
```
```suggestion
 * @param {string} password enter valid password
```

Fermain commented on src/js/api/auth/loginAuth.mjs:
```diff
@@ -0,0 +1,45 @@
+ * user log in. needs valid email and password.
+ * @param {text} email valid email
+ * @param {text} password enter valid password
+ * @param {url to api login endpoint} APIUrl url to api login endpoint
```
```suggestion
 * @param {string} APIUrl url to api login endpoint
```

Fermain reviewed (approved):
As you note in your intro, there are remaining UX concerns. The code is readable and does not contain errors or mistakes that I could discern. The app itself is attractive, responsive and delivers an uneven but good UX.

The core requirements are met and some of the stretch goals are also met. With this in mind I suggest focusing on the user journey. Make sure to add a redirect to the homepage if the user is not logged in so that they are bounced to the login page.

Fermain commented:
> Q: is the way i have added routing a good method, or is it unnecessary complicated?

I like this approach and it's similar to what you will see in React projects

StianSto commented:
## bug fixes and improvements
thank you so much for your feedback on Loom. i have spent some time trying to fix and improve my app in line with your feedback :) if you do have some time to spare someday, do take a look. there is no nned to rush with this of course :) i know you are a busy man, so im happy enough to just get this merged into main

#### bugs fixed: 

- filter not appending results
- post-options not working after re-renders (filter, posting etc...)

#### improvements: 
- added an "edit profile" on profile page
- user gets a simple feedback when updating their profile
- user gets feedback when creating a new post. also allows the user to instantly delete or go to edit post page
- user gets feedback when editing a post. 
- added delete post button in post-options
